docs/core/extensions/logger-message-generator.md

```diff
@@ -0,0 +1,320 @@
+
+```xml
+<PropertyGroup>
+  <LangVersion>9.0</LangVersion>
+</PropertyGroup>
+```
+
+#### Sample with exception as argument to log method
+
+Since the `ILogger.Log` API signature takes log level and optionally an exception per log call:
+
+```csharp
+public partial interface ILogger
+{
+    void Log<TState>(Microsoft.Extensions.Logging.LogLevel logLevel, Microsoft.Extensions.Logging.EventId eventId, TState state, System.Exception? exception, System.Func<TState, System.Exception?, string> formatter);
+}
+```
+
+Therefore, as a general rule, the first instance of `ILogger`, `LogLevel`, and `Exception` are treated specially in the log method signature of the source generator. Subsequent instances are treated like normal arguments to the message template:
+
+```csharp
+// below works
+[LoggerMessage(EventId = 110, Level = LogLevel.Debug, Message = "M1 {ex3} {ex2}")]
+static partial void LogMethod(ILogger logger, System.Exception ex, System.Exception ex2, System.Exception ex3);
+
+// but this warns:
+// DiagnosticSeverity.Warning - SYSLIB0025: Don't include a template for ex in the logging message since it is implicitly taken care of
+[LoggerMessage(EventId = 0, Level = LogLevel.Debug, Message = "M1 {ex} {ex2}")]
+static partial void LogMethod(ILogger logger, System.Exception ex, System.Exception ex2);
+```
+
+### Case-insensitive parameter/template name support
+
+The generator does case-insensitive comparison between parameters in message template and log message argument names so when the ILogger enumerates the state, the argument will be picked up by message template, which can make the logs nicer to consume:
+
+```csharp
+public partial class LoggingSample6
+{
+    private readonly ILogger _logger;
+
+    public LoggingSample6(ILogger logger)
+    {
+        _logger = logger;
+    }
+
+    [LoggerMessage(EventId = 10, Level = LogLevel.Information, Message = "Welcome to {City} {Province}!")]
+    public partial void LogMethodSupportsPascalCasingOfNames(string city, string province);
+
+    public void TestLogging()
+    {
+        LogMethodSupportsPascalCasingOfNames("Vancouver", "BC");
+    }
+}
+```
+
+Json console output (notice the log arguments become uppercase):
+```
+{
+  "EventId": 13,
+  "LogLevel": "Information",
+  "Category": "LoggingExample",
+  "Message": "Welcome to Vancouver BC!",
+  "State": {
+    "Message": "Welcome to Vancouver BC!",
+    "City": "Vancouver",
+    "Province": "BC",
+    "{OriginalFormat}": "Welcome to {city} {province}!"
+  }
+}
+```
+### Order of arguments do not matter
+
+There is no constraint in the ordering at this point. So the user could define `ILogger` as the last argument for example:
+```csharp
+[LoggerMessage(EventId = 110, Level = LogLevel.Debug, Message = "M1 {ex3} {ex2}")]
+static partial void LogMethod(System.Exception ex, System.Exception ex2, System.Exception ex3, ILogger logger);
```

Comment:
Will the generator here log `ex3 ex2` or `ex2 ex3`?
In LoggerMessage.Define only the position of the placeholder is used, not the name of the placeholder. So my question is: does the logger generator use the name of the placeholder?

If not this example should be changed or a sentence added about the behavior.
It may be astonishing to some users otherwise.

Reply:
thanks for the note.

Reply:
Good question @gfoidl

@maryamariyan and I tested this out, it is non-ordinal based - and matches the parameter names to the template placeholders. I'm adding details about this to the article.